src/mavlinkscenario.cpp

```diff
@@ -285,34 +285,42 @@ bool MavlinkScenario::add_onboard_message(const OnboardData &msg) {
      ****************************/    
 
     // timeseries
+    const std::string msg_name = "onboard log/" + msg.get_message_name() + "/";
+    const std::size_t stem_len = msg_name.size();
+    std::string fullname = msg_name;
     for (OnboardData::booldata_t::const_iterator dit = msg.get_booldata().begin(); dit != msg.get_booldata().end(); ++dit) {
         if (dit->first == "t") continue;
-        string fullname = "onboard log/" + msg.get_message_name() + "/" + dit->first;
+        fullname.resize(stem_len);
```

Comment:
Ok, that probably needs more explanation in the code:
- there is now only a single local `std::string` variable that is passed to all calls of `track_generic_timeseries` (before, multiple temporary strings were created for every single call, resulting in a lot of dynamic memory allocations).
- The main part, consisting of `"onboard log/" + msg.get_message_name() + "/"` stayes the same for all of those calls, but the last part (`dit->first`) has to be changed for every call
- The resize function SHORTENS the string such that only the constant (stem) part remains (the internal buffer/capacity of course stays the same size)
- The call to append then only copies a few new characters into the internal buffer space.

Instead of `resize` one could probably call `erase` to clarify the purpose.